core/lib/data/network/download/download_manager.dart

```diff
@@ -82,39 +81,28 @@ class DownloadManager {
     return streamController.stream.first;
   }
 
-  Future<bool> downloadFileForWeb(
-      String downloadUrl,
-      String filename,
-      String authentication) async {
+  bool createAnchorElementDownloadFileWeb(
+      Uint8List bytes,
+      String filename
+  ) {
     try {
-      final headerParam = Map<String, String>();
-      headerParam[HttpHeaders.authorizationHeader] = authentication;
-      headerParam[HttpHeaders.acceptHeader] = DioClient.jmapHeader;
-
-      http.Response res = await http.get(
-          Uri.parse(downloadUrl),
-          headers: headerParam);
+      final blob = html.Blob([bytes]);
+      final url = html.Url.createObjectUrlFromBlob(blob);
+      final anchor = html.document.createElement('a') as html.AnchorElement
+        ..href = url
+        ..style.display = 'none'
+        ..download = filename;
+      html.document.body?.children.add(anchor);
 
-      if (res.statusCode == 200) {
-        final blob = html.Blob([res.bodyBytes]);
-        final url = html.Url.createObjectUrlFromBlob(blob);
-        final anchor = html.document.createElement('a') as html.AnchorElement
-          ..href = url
-          ..style.display = 'none'
-          ..download = filename;
-        html.document.body?.children.add(anchor);
+      anchor.click();
 
-        anchor.click();
-
-        html.document.body?.children.remove(anchor);
-        html.Url.revokeObjectUrl(url);
-
-        return true;
-      }
+      html.document.body?.children.remove(anchor);
+      html.Url.revokeObjectUrl(url);
 
+      return true;
+    } catch (error) {
+      log('DownloadManager::createAnchorElementDownloadFileWeb(): ERROR: $error');
       return false;
-    } catch (exception) {
-      throw exception;
```

Comment:
why you not throw exception?

Reply:
I'm not handling any error events here at the moment. But maybe I'll add it so I can use it later.